### PR TITLE
Fix 8# issue-- Replace static "Object List" and "Object Creator" with names of objects?

### PR DESCRIPTION
**Pull request by serenabenny**
1.Fix 8# issue-- Replace static "Object List" and "Object Creator" with names of objects?
2.Added SetDefaultValue method to handle other types PrimaryKey or set default value.
3.Added IsCanEdit property in the UIOMaticFieldAttribute.
if set the value = false, just allow input on create time, didn't editable.
4.Fixed issue --The dropdown and checkboxlist view don't work with table and column names with a space in them.

[UIOMaticAttribute("People","icon-users","icon-user")]
[TableName("People")]
public class Person: IUIOMaticModel
{
    public Person() { }

    [UIOMaticIgnoreField]
    [PrimaryKeyColumn(AutoIncrement = true)]
    public int Id { get; set; }

    [UIOMaticField("First name","Enter the persons first name", IsCanEdit = false)]
    public string FirstName { get; set; }

    [UIOMaticField("Last name", "Enter the persons last name")]
    public string LastName { get; set; }

    [UIOMaticField("Picture", "Select a picture", View = "file")]
    public string Picture { get; set; }

    public override string ToString()
    {
        return FirstName + " " + LastName;
    }

    public IEnumerable<Exception> Validate()
    {
        var exs = new List<Exception>();

        if(string.IsNullOrEmpty(FirstName))
            exs.Add(new Exception("Please provide a value for first name"));

        if (string.IsNullOrEmpty(LastName))
            exs.Add(new Exception("Please provide a value for last name"));

        return exs;
    }
	public void SetDefaultValue()
	{
		if(id==0)
			id=newID();
		if (string.IsNullOrWhiteSpace(Status))
		{
			Status = "1";
		}
		if (CreatedDateTime == default(DateTime))
		{
			CreatedDateTime = DateTime.Now;
			CreatedBy = GetCurrentUserId();
		}
		UpdatedDateTime = DateTime.Now;
		UpdatedBy = GetCurrentUserId();
	}
}

### CODE REVIEW

**Comment by TimGeyssens** [comment]
Wow you are on a roll  :+1:  great idea with the set default value

File: src/UIOMatic/Controllers/PetaPocoObjectController.cs [hunk 1]
```diff
@@ -449,17 +545,18 @@ public object PostUpdate(ExpandoObject objectToUpdate)
 
 
             }
-
+            ((IUIOMaticModel)ob).SetDefaultValue();
```
Comment (TimGeyssens) [inline src/UIOMatic/Controllers/PetaPocoObjectController.cs]: Shouldn't the set default value only run on insert?

**Comment by TimGeyssens** [comment]
For the default value, maybe an event model might be better no? Since now the code runs on both insert and update...

**Comment by serenabenny** [comment]
I think,Just adding a call into the POSTUPDATE method is simple.

**Comment by TimGeyssens** [comment]
@serenabenny I implemented the events on the latest release https://github.com/TimGeyssens/UIOMatic/releases/tag/1.4.0

**Comment by TimGeyssens** [comment]
@serenabenny any chance you can submit these as separate pull request since now it's one big one already containing stuff that has been added? That would amazing, thanks!

**Comment by serenabenny** [comment]
Sorry , I don't know how to separate pull requests, Are you have any any manual for this? 
I merged your code already!

**Comment by serenabenny** [comment]
I checked Even Model, not a good of use, if I have 100 custom tables and to set even, then this will make the other non-related Peco's activation event on Create & Update method.

**Comment by TimGeyssens** [comment]
@serenabenny ok will do it manually then but would be ace if you have other ones in the pipe line to do small pull requests with 1 bug fix/feature instead of a big one :)

**Comment by TimGeyssens** [comment]
All of your fixes have been merged manually, thanks :)

**Comment by TimGeyssens** [comment]
Will close now as I'm manually merging the changes

**Pull request closed without merging**